homeassistant/components/mqtt/light/schema_json.py

```diff
@@ -594,13 +590,13 @@ async def async_turn_on(self, **kwargs: Any) -> None:  # noqa: C901
         self._set_flash_and_transition(message, **kwargs)
 
         if ATTR_BRIGHTNESS in kwargs and self._config[CONF_BRIGHTNESS]:
-            brightness_normalized = kwargs[ATTR_BRIGHTNESS] / DEFAULT_BRIGHTNESS_SCALE
-            brightness_scale = self._config[CONF_BRIGHTNESS_SCALE]
-            device_brightness = min(
-                round(brightness_normalized * brightness_scale), brightness_scale
-            )
             # Make sure the brightness is not rounded down to 0
-            device_brightness = max(device_brightness, 1)
+            device_brightness = max(
+                round(
+                    self._config[CONF_BRIGHTNESS_SCALE] * kwargs[ATTR_BRIGHTNESS] / 255
+                ),
+                1,
+            )
```

Comment:
Why is this changed? The change is not covered by changes in tests.

Reply:
It is based on de fan percentage scaling code we use for fan. In fact the logic is not changed. Therefor the tests did not need an update.

Reply:
removed, I'll open a PR for some helpers, so anyone can use them